cmd/ionctl/root/root.go

```diff
@@ -0,0 +1,68 @@
+package root
+
+import (
+	"fmt"
+	"os"
+
+	homedir "github.com/mitchellh/go-homedir"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+var cfgFile string
+var ManagementApiEndpoint string
+
+// RootCmd represents the base command when called without any subcommands
+var RootCmd = &cobra.Command{
+	Use:   "ionclt",
+	Short: "ionctl lets you easily work with the ion system",
+}
+
+// Execute adds all child commands to the root command and sets flags appropriately.
+// This is called by main.main(). It only needs to happen once to the rootCmd.
+func Execute() {
+	if err := RootCmd.Execute(); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+}
+
+func init() {
+	cobra.OnInitialize(initConfig)
+
+	// Here you will define your flags and configuration settings.
+	// Cobra supports persistent flags, which, if defined here,
+	// will be global for your application.
+	RootCmd.PersistentFlags().StringVar(&cfgFile, "config", "", "config file (default is $HOME/.ioncli.yaml)")
+	RootCmd.PersistentFlags().StringVar(&ManagementApiEndpoint, "endpoint", "localhost:9000", "The Mangement api endpoint to use")
+
+	// Cobra also supports local flags, which will only run
+	// when this action is called directly.
+	RootCmd.Flags().BoolP("toggle", "t", false, "Help message for toggle")
+}
+
+// initConfig reads in config file and ENV variables if set.
+func initConfig() {
+	if cfgFile != "" {
+		// Use config file from the flag.
+		viper.SetConfigFile(cfgFile)
+	} else {
+		// Find home directory.
+		home, err := homedir.Dir()
+		if err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+
+		// Search config in home directory with name ".ioncli" (without extension).
+		viper.AddConfigPath(home)
+		viper.SetConfigName(".ioncli")
```

Comment:
we'll need to be consistent `.ioncli` or `.ionctl`?

Reply:
Yeah good point, I flip flopped on the naming a bit. Do you think it's better to go as ionctl or ioncli?

Reply:
Or I just call the cli tool `ion` to make things either simpler

Reply:
I'd rather go with brevity for a CLI - `ion` is better. We'd end up aliasing `ionctl` to `i` or `ion` anyway. We can refer to it as ion CLI in docs etc as it's clearer. We don't want the `kubectl` debate on how the hell you say it!

Reply:
Yeah that makes sense - lets got for `ion`